source/lesson-2-ruby-and-the-command-line.html.md.erb

```diff
@@ -0,0 +1,722 @@
+pressing enter. Do you get the right answer?
+
+In ruby, the `+` operator adds numbers together. Other operators include:
+
+* `-` - subtract
+* `*` - multiply
+* `/` - divide
+* `**` - raise to the power of
+
+You can also use brackets (`()`) to group expressions.
+
+#### Task 2: Maths challenge
+
+Use `irb` to work out the answer to "191 multiplied by 7".
+
+<details>
+<summary>Answer</summary>
+
+```
+irb(main):016:0> 191 * 7
+=> 1337
+```
+
+</details>
+
+### Strings
+
+In the real world strings tie things up. Programming strings have *nothing*
+to do with real-world strings.
+
+Programming strings are used to store collections of letters and numbers.
+That could be a single letter like `"a"`, a word like `"hi"`, or a sentence like
+`"Hello my friends."`.
+
+A Ruby string is written as a quote (`"`) followed by some letters, numbers,
+or symbols and ended with a closing quote (`"`). The shortest possible string
+is called the empty string: `""`. It’s not uncommon for a single string to
+contain paragraphs or even pages of text.
+
+If you type a string into `irb` it will echo it back at you:
+
+```
+irb(main):017:0> "Hello IRB"
+=> "Hello IRB"
+```
+
+#### Getting things wrong in IRB
+
+Just like in the command line, if you give `irb` a command it doesn't
+understand it will show you an error message. Again, don't panic! Everything
+is fine.
+
+`irb` will do its best to explain why it didn't understand your ruby, but
+sometimes the error messages can be hard for a beginner to understand.
+
+For example:
+
+```
+irb(main):018:0> "Hello IRB" + 7
+TypeError: no implicit conversion of Fixnum into String
+        from (irb):1:in `+'
+        from (irb):1
+        from /usr/bin/irb:11:in `<main>'
+```
+
+Here, `irb` is telling us that you're not allowed to add strings and numbers
+together in ruby.
+
+### Variables
+
+Programming is all about creating abstractions, and in order to create an
+abstraction we must be able to assign names to things. Variables are a way of
+creating a name for a piece of data.
+
+Creating and using variables in ruby is super simple, it's just `name` `=`
+`expression`. Some examples:
+
+```ruby
+name = "Fido"
+age_human_years = 4
+age_dog_years = age_human_years * 7
+```
+
+Variable names in ruby have to start with a letter, and they can't contain
+spaces or "special" characters like `-`, `$`, `@` and `&`.
+
+As a style convention, ruby variables use underscores to separate the bits of
+the name - this is called `snake_case` (as opposed to `camelCase` or `PascalCase` which are used elsewhere).
+
+### String interpolation
+
+We often need to build a string out of other strings and ruby expressions. In
+ruby we can do this with "string interpolation".
+
+Within the string we use the interpolation marker `#{}`. Inside those
+brackets we can put any variables or Ruby code which will be evaluated,
+converted to a string, and output in that spot of the outer string.
```

Comment:
One thing that threw me when learning ruby was the difference between a single quote and a double quote in string interpolation. Do you think it's worth mentioning the difference, or would it be best left out as it adds more complexity?

Reply:
I've attempted to leave out the fact that you can use single quotes completely to avoid tripping people up. In general I'm trying to avoid any language feature that doesn't absolutely _need_ to be introduced (see also, optional return statements, optional brackets on methods).